### PR TITLE
Remove ziplinks and some cleanup

### PR DESCRIPTION
**Pull request by ariscop**
adds complexity and latency for a 50% bandwidth saving
given irc is not a bandwidth hog this doesn't seem to be a useful feature

add documentation for ssld's ipc

### CODE REVIEW

File: src/sslproc.c [hunk 1]
```diff
@@ -479,6 +442,7 @@ ssl_read_ctl(rb_fde_t * F, void *data)
     rb_setselect(ctl->F, RB_SELECT_READ, ssl_read_ctl, ctl);
 }
 
+/* Return the ssld with the fewest clients */
```
Comment (Xe) [inline src/sslproc.c]: I didn't know ratbox does this. Clever.

File: ssld/ssld.c [hunk 2]
```diff
@@ -19,13 +19,70 @@
  *  USA
  */
 
+/**
+ * IPC format
```
Comment (Xe) [inline ssld/ssld.c]: This probably belongs in the [wiki](https://github.com/Elemental-IRCd/elemental-ircd/wiki) too.

File: ssld/ssld.c [hunk 3]
```diff
@@ -112,40 +159,36 @@ typedef struct _conn {
...
 
-#define FLAG_SSL	0x01
-#define FLAG_ZIP	0x02
-#define FLAG_CORK	0x04
-#define FLAG_DEAD	0x08
-#define FLAG_SSL_W_WANTS_R 0x10	/* output needs to wait until input possible */
-#define FLAG_SSL_R_WANTS_W 0x20	/* input needs to wait until output possible */
-#define FLAG_ZIPSSL	0x40
+#define FLAG_SSL           0x01
+//define FLAG_ZIP          0x02 /* Removed */
```
Comment (Xe) [inline ssld/ssld.c]: Git saves things, can just `dd` this line.

Comment (ariscop) [inline ssld/ssld.c]: left it in to explain the gap between ssl and cork, and didn't want to change all the flags to avoid the gap

Comment (Xe) [inline ssld/ssld.c]: ah, makes sense.

**Comment by Xe** [comment]
:+1:

**Pull request merged**